File: retrofit2_rxjava2/src/main/java/jack/retrofit2_rxjava2/model/ApiResponse.java
package jack.retrofit2_rxjava2.model;

/**
 * created by Jack
 * email:yucrun@163.com
 * date:19-5-5
 * describe:服务器返回的json数据转化成bean格式;
 */
public class ApiResponse<T> {

    private int errorCode;
    private String errorMsg;
    private T data;

    public int getErrorCode() {
        return errorCode;
    }

    public void setErrorCode(int errorCode) {
        this.errorCode = errorCode;
    }

    public String getErrorMsg() {
        return errorMsg;
    }

    public void setErrorMsg(String errorMsg) {
        this.errorMsg = errorMsg;
    }

    public T getData() {
        return data;
    }

    public void setData(T data) {
        this.data = data;
    }

    @Override
    public String toString() {
        return "ApiResponse{" +
                "errorCode=" + errorCode +
                ", errorMsg='" + errorMsg + '\'' +
                ", data=" + data +
                '}';
    }

}

File: dir_basic/library_common_business/src/main/java/cn/jack/library_common_business/entiy/ApiResponse.kt
package cn.jack.library_common_business.entiy

/**
 * created by Jack
 * date:19-5-5
 * describe:服务器返回的json数据转化成bean格式;
 */
class ApiResponse<T> {
    var errorCode = 0
    var errorMsg: String? = null
    var data: T? = null
        private set

    override fun toString(): String {
        return "ApiResponse{" +
                "errorCode=" + errorCode +
                ", errorMsg='" + errorMsg + '\'' +
                ", data=" + data +
                '}'
    }
}